docs/tutorials/typescript/work-queue-slack-app/index.mdx

```diff
@@ -0,0 +1,991 @@
+### Dependencies for the projects
+
+**Temporal Application dependencies**
+
+In the `temporal-application` project, you will need the following dependencies in your `package.json`:
+
+```json
+{
+  // ...
+  "dependencies": {
+    "@temporalio/worker": "^1.10.1",
+    "dotenv": "^16.1.3",
+    "path": "^0.12.7"
+  },
+  "devDependencies": {
+    "@types/node": "^20.14.7",
+    "ts-node": "^10.9.2",
+    "typescript": "^5.4.5"
+  }
+  // ...
+}
+```
+
+**Slack bot dependencies**
+
+In the `bot` project, you will need the following dependencies in your `package.json`:
+
+```json
+{
+  // ...
+  "dependencies": {
+    "@slack/bolt": "^3.19.0",
+    "@slack/web-api": "^6.12.0",
+    "@temporalio/client": "^1.7.4",
+    "crypto": "^1.0.1",
+    "date-fns": "^3.6.0",
+    "dotenv": "^16.0.3"
+  },
+  "devDependencies": {
+    "@types/node": "^20.14.7",
+    "ts-node": "^10.9.2",
+    "typescript": "^5.4.5"
+  }
+  // ...
+}
+```
+
+### Environment variables
+
+First, make sure you have `.env` added to your `.gitignore` file to avoid committing your credentials.
+
+**Slack bot environment variables**
+
+Grab the [Slack credentials](#credentials) you got from the Slack App configuration and add them to a `.env` file in the `bot` project:
+
+```plaintext
+SLACK_SIGNING_SECRET="<slack-signing-secret>"
+SLACK_BOT_TOKEN="<slack-bot-token>"
+SLACK_APP_TOKEN="<slack_app_token>"
+SLACK_WORKSPACE="<slack_workspace>"
+ENV="dev"
+```
+
+Also add your Slack Workspace and an `ENV` variable.
+The `ENV` variable is prepended to the Task Queue name for your Temporal Application.
+This enables you to use the same Temporal Namespace locally and in production without worrying about production Tasks getting executed on your local machine.
+
+**Temporal Application environment variables**
+
+In the `temporal-application` project, make sure you have a `.env` file and for now, just add the `ENV` variable:
+
+```plaintext
+ENV="dev"
+```
+
+### Define commone types
+
+Create a file `types.ts` in the `common-types` directory and define a Work Item data object and its statuses:
+
+<!--SNIPSTART typescript-slack-app-common-types-->
+
+[docs/tutorials/typescript/work-queue-slack-app/code/common-types/types.ts](https://github.com/temporalio/temporal-learning/blob/edu-2636-workqueue-slackbot-tut/docs/tutorials/typescript/work-queue-slack-app/code/common-types/types.ts)
+
+```ts
+export interface WorkqueueData {
+  id: string;
+  timestamp: string;
+  channelName: string;
+  channelId: string;
+  userId: string;
+  work: string;
+  status: WorkqueueStatus;
+  claimantId?: string;
+  // Add more properties as needed
+}
+
+export enum WorkqueueStatus {
+  Backlog = 1,
+  InProgress = 2,
+  Done = 3,
+}
+```
+
+<!--SNIPEND-->
+
+## Work Queue Workflow
+
+Before you start building the Slack bot, you need to create a Temporal Workflow to persist the state of the Work Queue.
+
+This Workflow will be "long living" and any given instance of it will map directly to a specific Slack channel.
```

Comment:
I would not scare quote this -- explain it either with ellipses or a link that explains the Entity Workflow pattern as it pertains to Temporal.